core/src/main/java/org/elasticsearch/search/sort/SortBuilder.java

```diff
@@ -55,34 +54,24 @@
     protected SortOrder order = SortOrder.ASC;
     public static final ParseField ORDER_FIELD = new ParseField("order");
 
-    private static final Map<String, SortBuilder<?>> PARSERS;
+    private static final Map<String, Parser<?>> PARSERS;
+
+    @FunctionalInterface
+    public interface Parser<T extends SortBuilder<?>> {
```

Comment:
I don't think BiFunction can throw and I don't think it is worth catching and wrapping here. I sure can make this private though!